force-app/main/default/classes/SOQL.cls

```diff
@@ -50,22 +52,32 @@ public with sharing class SOQL {
      * @return A list of Contact Ids.
      * Update the corresponding test class and testGetContactIdsByLastName method to cover this method
      */
-    //[public | private ] [static] data_type method_name (input parameters) {
-    //  The body of the method
-    //}
+    
+        public static List<Id> retrieveContactIds(String lastName1, String lastName2) {
+            
+           List<Id> contactIds = new List<Id>();
+            List<Contact> gatherContacts = [SELECT Id FROM Contact WHERE LastName = :LastName1 OR LastName = :LastName2 WITH SECURITY_ENFORCED];
+            for(Contact conId: gatherContacts){
+                contactIds.add(conId.Id);
+            }
+            return contactIds;
+    
+    }
+
 
     /**
      * Question 4
-     * Update/Uncomment the template code to create a method
+     * Update/Uncomment the template code to create a method 
      * Returns List of closed won opportunities for a specific Account.
      * Example: yourMethodName('001XXXXXXXXXXXX') should return the list of closed won opportunities for the Account with the Id
      * @param accountId The ID of the Account.
      * @return A list of Opportunities with important opportunities fields
      * Update the corresponding test class and testGetClosedWonOpportunityCountByAccount method to cover this method
      */
-    //[public | private ] [static] data_type method_name (input parameters) {
-    //  The body of the method
-    //}
+            public static List<Opportunity> retrieveOpportunityFields(Id accountId) {
+                Opportunity[] opportunityQuery = [SELECT Name, StageName, Amount, Type, CloseDate, Opportunity.AccountId FROM Opportunity WHERE StageName = 'Closed Won' AND AccountId = :accountId WITH SECURITY_ENFORCED];
+                    return opportunityQuery;
```

Comment:
This is the first place where you'll want to start thinking about query formatting. As your queries get bigger and more complex, you will want to start formatting them vertically so they are easier to read.
```suggestion
        List<Opportunity> opps = [
                SELECT 
                        Id,
                        Name,
                        StageName,
                        AccountId,
                        CloseDate,
                        Amount 
                FROM Opportunity 
                WHERE AccountId = :accId 
                AND StageName = 'Closed Won'
        ];
        return opps;
```

Additionally, `opportunityQuery` doesn't really make sense as a variable name. The query is the string on line 78, you are returning Opportunity SObjects, records. So you would not call it a query.